### DISCUSSION TITLE
编译中有依赖错误

### DISCUSSION THREAD

**Discussion by yyshell** [immortalwrt/immortalwrt#1235] [Q&A]
### Describe the bug

编译中出现难以解决的依赖问题，导致出错。

### ImmortalWrt version

lastest snapshot

### ImmortalWrt release

lastest snapshot

### ImmortalWrt target/subtarget

lastest snapshot

### Device

x86-64

### Image kind

Self-built image

### Steps to reproduce

Collected errors:
 * pkg_hash_check_unresolved: cannot find dependency libubox20240126 for libuci20130104
 * pkg_hash_check_unresolved: cannot find dependency libubox20240126 for libubus20231128
 * satisfy_dependencies_for: Cannot satisfy the following dependencies for base-files:
 * 	libubox20240126
 * opkg_install_cmd: Cannot install package base-files.
 * pkg_hash_check_unresolved: cannot find dependency libubox20240126 for libuclient20201210
 * satisfy_dependencies_for: Cannot satisfy the following dependencies for ipv6helper:
 * 	libubox20240126
 * opkg_install_cmd: Cannot install package ipv6helper.
 * pkg_hash_check_unresolved: cannot find dependency libubox20240126 for libustream-openssl20201210
make[2]: *** [package/Makefile:71: package/install] Error 255
make[2]: Leaving directory '/home/adam/immortalwrt'
make[1]: *** [package/Makefile:124: /home/adam/immortalwrt/staging_dir/target-x86_64_musl/stamp/.package_install] Error 2
make[1]: Leaving directory '/home/adam/immortalwrt'
make: *** [/home/adam/immortalwrt/include/toplevel.mk:233: world] Error 2

### Actual behaviour

build failed

### Expected behaviour

build successfully

### Additional info

no

### Diffconfig

```text
#!/bin/sh
make ./scripts/config/conf >/dev/null || { make ./scripts/config/conf; exit 1; }
grep \^CONFIG_TARGET_ .config | head -n3 > tmp/.diffconfig.head
grep \^CONFIG_TARGET_DEVICE_ .config >> tmp/.diffconfig.head
grep '^CONFIG_ALL=y' .config >> tmp/.diffconfig.head
grep '^CONFIG_ALL_KMODS=y' .config >> tmp/.diffconfig.head
grep '^CONFIG_ALL_NONSHARED=y' .config >> tmp/.diffconfig.head
grep '^CONFIG_DEVEL=y' .config >> tmp/.diffconfig.head
grep '^CONFIG_TOOLCHAINOPTS=y' .config >> tmp/.diffconfig.head
grep '^CONFIG_BUSYBOX_CUSTOM=y' .config >> tmp/.diffconfig.head
grep '^CONFIG_TARGET_PER_DEVICE_ROOTFS=y' .config >> tmp/.diffconfig.head
./scripts/config/conf --defconfig=tmp/.diffconfig.head -w tmp/.diffconfig.stage1 Config.in >/dev/null
./scripts/kconfig.pl '>+' tmp/.diffconfig.stage1 .config >> tmp/.diffconfig.head
./scripts/config/conf --defconfig=tmp/.diffconfig.head -w tmp/.diffconfig.stage2 Config.in >/dev/null
./scripts/kconfig.pl '>' tmp/.diffconfig.stage2 .config >> tmp/.diffconfig.head
cat tmp/.diffconfig.head
rm -f tmp/.diffconfig tmp/.diffconfig.head
```

### Terms

- [X] I am reporting an issue for ImmortalWrt, not an unsupported fork.

**Comment by vpei** [comment]
删除重新编译应该就没问题。

**Reply by yyshell** [reply]
已试，没用的。上面有个讨论，在线构建，类似的错误。

**Comment by xlighting2017** [comment]
> 已试，没用的。上面有个讨论，在线构建，类似的错误。

https://github.com/openwrt/openwrt/issues/15113

**Comment by yyshell** [comment]
最近几次合并代码后，内核升级到6.6，此问题解决了。
